apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -80,49 +80,40 @@ export const commands = {
   // current or previous frame.
   anySpeechIncludesValues(currentVariables, previousVariables) {
     const spriteIds = this.getSpriteIdsInUse();
-    let result = false;
-    const values = previousVariables
-      ? Object.values(currentVariables).concat(Object.values(previousVariables))
-      : Object.values(currentVariables);
-    for (let i = 0; i < spriteIds.length; i++) {
-      values.forEach(value => {
-        const speechText = this.getLastSpeechBubbleForSpriteId(
-          spriteIds[i]
-        )?.text;
-        const type = typeof speechText;
-        // We only want to set result to true here, so that any positive test
-        // allows the overall criterion to pass.
-        switch (type) {
-          case 'string':
-            if (speechText.includes(value)) {
-              result = true;
-            }
-            break;
-          case 'number':
-            if (speechText === value) {
-              result = true;
-            }
-            break;
-          default:
-            break;
+    const values = Object.values(currentVariables);
+    if (previousVariables) {
+      values.push(...Object.values(previousVariables));
+    }
+
+    for (const spriteId of spriteIds) {
+      const speechText = this.getLastSpeechBubbleForSpriteId(spriteId)?.text;
+      if (speechText) {
+        for (const value of values) {
+          if (typeof speechText === 'string' && speechText.includes(value)) {
+            return true;
+          } else if (typeof speechText === 'number' && speechText === value) {
```

Comment:
If you're looking for this to return true in the following condition
```
Return true if any sprite's speech include values from the given object(s)
```
then why is this checking `speechText === value` for numerical values? Actually, I guess my real question is, why is this called `speechText` if it's sometimes numerical?

I'm wondering if we could do something like...
```
for (const spriteId of spriteIds) {
  const speechText = this.getLastSpeechBubbleForSpriteId(spriteId)?.text;
  if (speechText && values.some(value => 
    `${speechText}`.includes(`${value}`)
  )) {
    return true;
  }
}
return false;
```